### ISSUE TITLE
Run FastRCNN model

### ISSUE THREAD

**Issue by ebzkurt1** [Process-Dynamics/LiveX_inference#1]
This is an initial issue regarding the old FastRCNN model. We have the code and the model file, so the idea is running it on our data.

- [x] Get a simple result, doesn't have to be good. Just make sure that we can use the model.
- [ ] Share the inference result and discuss whether it needs a new training

**Comment by ebzkurt1** [comment]
I have managed to run the model myself using the jupyter notebook, now, I will try to use our own data with the model

**Comment by Piersman** [comment]
Great news!

Is that running on the Jetson Orin?

**Comment by Shun-Yang-OX** [comment]
A request has been made to pull the relevant files for the Faster R-CNN Model, including the **inference** script, the **test set** images, and the corresponding **README** file. The trained model file is available via the links provided in the README. This model has not yet been tested on Jetson, and it is the V1 version, but it can be utilised for subsequent input research. (The trained V2 model is on Enzo's Linux server. Due to a brief power outage in the department last weekend, the V2 model will be uploaded once IT has restored connectivity to the Linux server.)

**Comment by ebzkurt1** [comment]
Great, you can create the Pull request if you want to

**Comment by ebzkurt1** [comment]
I have also managed to run the model using the guide @Shun-Yang-OX written. 

The test data is also in the repo and steps are straight forward. I will try to get some inference metrics and share them now. I especially want to get the GPU memory usage of the data.

**Comment by Piersman** [comment]
I spent some time this week working on the inference for the beam time.

Managed to get the code uploaded by @Shun-Yang-OX working on our DAQ machines, reaching about ~8 frames per second during the model inferencing with the P4000 GPU that we've currently got.


Here's the GPU  while running the inference script that was provided.
``` bash
Every 2.0s: nvidia-smi                                                                                                                                                                                                                       te7aegdev13.te.rl.ac.uk: Fri Sep  5 14:13:53 2025

Fri Sep  5 14:13:53 2025
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.133.20             Driver Version: 570.133.20     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Quadro P4000                   Off |   00000000:01:00.0 Off |                  N/A |
| 50%   49C    P0             91W /  105W |     899MiB /   8192MiB |     84%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  Quadro P400                    Off |   00000000:41:00.0 Off |                  N/A |
| 34%   35C    P8            N/A  /  N/A  |       5MiB /   2048MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A          824013      C   python                                  894MiB |
+-----------------------------------------------------------------------------------------+
```


When running against frames from the Orca Quest camera (2304x4096) the GPU memory usage goes up to around ~1250MB

**Comment by Piersman** [comment]
Here's nvidia-smi while the inference is running against frames in odin-data:

``` bash
Every 2.0s: nvidia-smi                                                                                                                                                                                                                       te7aegdev13.te.rl.ac.uk: Fri Sep  5 14:18:01 2025

Fri Sep  5 14:18:02 2025
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.133.20             Driver Version: 570.133.20     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Quadro P4000                   Off |   00000000:01:00.0 Off |                  N/A |
| 47%   45C    P0             44W /  105W |    1153MiB /   8192MiB |     82%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  Quadro P400                    Off |   00000000:41:00.0 Off |                  N/A |
| 34%   34C    P8            N/A  /  N/A  |       5MiB /   2048MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A          826169      C   python                                 1148MiB |
+-----------------------------------------------------------------------------------------+
```

And the output from the python inference script:
``` bash
Processing frame 606
    Tensor shape: torch.Size([3, 2304, 4096])
    Tensor dtype: torch.float32
    Tensor device: cuda:0
    Tensor range: [0.000, 1.000]
  Found 18 objects (inference time: 135.9ms)
  Min: 0, Max: 65535, Mean: 17479.87
  Updated results: Frame 606, Total objects: 18
```

We've piped the example images provided in the repo through odin-data, so are confident that the inference is working as expected as the detected objects seems to match the other results. We're currently working on having a method for applying corrections to the images before inference takes place.

**Comment by ebzkurt1** [comment]
Wow! That is actually great. So, we can run the model during the beamtime i guess.

Are there anything left for us to do in that regard and can you share the code?

**Comment by Piersman** [comment]
> So, we can run the model during the beamtime i guess.

That's the hope, once I get the image corrections working I'll push a copy of the code here.

I can't think of anything left for us to work on, unless we want to try and run the liquid segmentation model at the beam time?
Although I'm sure we'll have plenty of time to work on that during the beam time itself.

**Comment by ebzkurt1** [comment]
Perfect! Thank's a lot.

**Comment by ebzkurt1** [comment]
Can you share the code so that we can run inference on the old data? We want to get some type of analysis to observe the model results and share them with the team.

**Comment by Piersman** [comment]
I'll get the code uploaded once the image correction functionality is working correctly.

However, for python code to work it's required to run alongside odin-data and is only designed for inferencing on live data before being saved out into a .h5 file. So we'll probably need another method for inferencing on data that's already been saved out to .h5 files.

**Comment by ebzkurt1** [comment]
We can try to work our way from your code and write a different version geared towards inferencing old data. I am guessing that it will need a different dataloader but the inference part should stay the same. or at least i hope :D

**Comment by Shun-Yang-OX** [comment]
The download link for the V2 model has been added to the README.md file, and minor modifications have been made to the inference script to support the V2 model.
